go/tools/builders/embed.go

```diff
@@ -0,0 +1,240 @@
+	key := filepath.FromSlash(strings.TrimPrefix(filename, workspacePrefix))
+	if c.flatten {
+		key = path.Base(filename)
+	}
+	return strconv.Quote(key)
+}
+
+func run(args []string) error {
+	c := newConfiguration(args)
+
+	f, err := os.Create(c.out)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	w := bufio.NewWriter(f)
+	defer w.Flush()
+
+	tpl := template.Must(template.New("embed").Parse(headerTpl))
+	err = tpl.Execute(w, c)
+	if err != nil {
+		return err
+	}
+
+	dataBegin, dataEnd := "\"", "\"\n"
+	if !c.strData {
+		dataBegin, dataEnd = "[]byte(\"", "\")\n"
+	}
+
+	if !c.Multi {
+		_, err := fmt.Fprintf(w, "var %s = %s", c.Var, dataBegin)
+		if err != nil {
+			return err
+		}
+		err = embedFileContents(w, c.Sources[0])
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprint(w, dataEnd)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if len(c.Sources) == 0 {
+		return nil
+	}
+
+	_, err = fmt.Fprint(w, "var (\n")
+	if err != nil {
+		return err
+	}
+	for i, filename := range c.Sources {
+		_, err := fmt.Fprintf(w, "\t%s_%d = %s", c.Var, i, dataBegin)
+		if err != nil {
+			return err
+		}
+		err = embedFileContents(w, filename)
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprint(w, dataEnd)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = fmt.Fprint(w, ")\n")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func newConfiguration(args []string) *configuration {
+	var c configuration
+	flags := flag.NewFlagSet("embed", flag.ExitOnError)
+	flags.StringVar(&c.Label, "label", "", "Label of the rule being executed (required)")
+	flags.StringVar(&c.Package, "package", "", "Go package name (required)")
+	flags.StringVar(&c.Var, "var", "", "Variable name (required)")
+	flags.BoolVar(&c.Multi, "multi", false, "Whether the variable is a map or a single value")
+	flags.StringVar(&c.out, "out", "", "Go file to generate (required)")
+	flags.StringVar(&c.workspace, "workspace", "", "Name of the workspace (required)")
+	flags.BoolVar(&c.flatten, "flatten", false, "Whether to access files by base name")
+	flags.BoolVar(&c.strData, "string", false, "Whether to store contents as strings")
+	flags.Parse(args[1:])
+	if c.Label == "" {
+		log.Fatal("error: -label option not provided\n")
+	}
+	if c.Package == "" {
+		log.Fatal("error: -package option not provided\n")
+	}
+	if c.Var == "" {
+		log.Fatal("error: -var option not provided\n")
+	}
+	if c.out == "" {
+		log.Fatal("error: -out option not provided\n")
+	}
+	if c.workspace == "" {
+		log.Fatal("error: -workspace option not provided\n")
+	}
+	c.Sources = flags.Args()
+	if !c.Multi && len(c.Sources) != 1 {
+		log.Fatalf("error: -multi flag not given, so want exactly one source; got %d", len(c.Sources))
+	}
+	return &c
+}
+
+func embedFileContents(w io.Writer, filename string) error {
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	r := bufio.NewReader(f)
+
+	w = &escapeWriter{w}
```

Comment:
inline both of these to just

_, err = io.Copy(&escapeWriter{w}, bufio.NewReader(f))